src/plugins/presentation_panel/public/panel_actions/customize_panel_action/custom_time_range_badge.test.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Filter, TimeRange, type AggregateQuery, type Query } from '@kbn/es-query';
+
+import { PublishesLocalUnifiedSearch } from '@kbn/presentation-publishing';
+import { BehaviorSubject } from 'rxjs';
+import { CustomTimeRangeBadge } from './custom_time_range_badge';
+
+const mockTimeRange: TimeRange = { from: 'now-17m', to: 'now' };
+
+describe('custom time range badge action', () => {
+  let action: CustomTimeRangeBadge;
+  let context: { embeddable: PublishesLocalUnifiedSearch };
+
+  let updateTimeRange: (timeRange: TimeRange | undefined) => void;
+
+  beforeEach(() => {
+    const timeRangeSubject = new BehaviorSubject<TimeRange | undefined>(undefined);
+    updateTimeRange = (timeRange) => timeRangeSubject.next(timeRange);
+
+    action = new CustomTimeRangeBadge();
+    context = {
+      embeddable: {
+        localTimeRange: timeRangeSubject,
+        localFilters: new BehaviorSubject<Filter[] | undefined>(undefined),
+        localQuery: new BehaviorSubject<Query | AggregateQuery | undefined>(undefined),
+      },
+    };
+  });
+
+  it('is compatible when api has a time range', async () => {
+    updateTimeRange(mockTimeRange);
+    expect(await action.isCompatible(context)).toBe(true);
+  });
+
+  it('is incompatible when api is missing required functions', async () => {
+    const emptyContext = { embeddable: {} };
+    expect(await action.isCompatible(emptyContext)).toBe(false);
+  });
+
+  it('calls onChange when time range changes', () => {
+    const onChange = jest.fn();
+    updateTimeRange(mockTimeRange);
+    updateTimeRange(undefined);
+    action.subscribeToCompatibilityChanges(context, onChange);
```

Comment:
Move updates to after subscriptions is created

Reply:
Done!